aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -597,6 +603,65 @@ public TransferObserver upload(String key, File file, ObjectMetadata metadata, C
         return upload(getDefaultBucketOrThrow(), key, file, metadata, cannedAcl, listener);
     }
 
+    /**
+     * Starts uploading the inputStream to the <b>default</b> bucket, using the given key.
+     *
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     * @return A TransferObserver used to track upload progress and state
+     */
+    public TransferObserver upload(String key, InputStream inputStream) throws IOException {
+        // Saves the data as a file in the temporary directory
+        File file = File.createTempFile(UUID.randomUUID().toString(), ".tmp");
+        if (inputStream != null) {
+            try {
+                OutputStream outStream = new FileOutputStream(file);
+                byte[] buffer = new byte[1024];
+                int bytesRead;
+                // Keep reading until reaches the end of the stream
+                while ((bytesRead = inputStream.read(buffer)) != -1) {
+                    outStream.write(buffer, 0, bytesRead);
+                    outStream.flush();
+                }
+                inputStream.close();
+                outStream.close();
+            } catch (Exception e) {
+                file = null;
+            }
+        }
+        return upload(key, file);
+    }
+
+    /**
+     * Starts uploading the inputStream to the given bucket, using the given key.
+     *
+     * @param bucket        The name of the bucket to upload the new object to.
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     * @return A TransferObserver used to track upload progress and state
+     */
+    public TransferObserver upload(String bucket, String key, InputStream inputStream) throws IOException {
```

Comment:
looks like `IOException` gets thrown if it fails to create a temp file? Maybe we should just hide this implementation detail and throw a customized error?